sources/packages/backend/libs/services/src/disbursement-schedule/disbursement-schedule-service.ts

```diff
@@ -260,43 +252,39 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
   }
 
   /**
-   * Rollback overawards from the current and future applications.
+   * Rollback overawards from the current application being reassessed.
    * Past pending assessment will not be impacted.
-   * @param studentId student id.
-   * @param offeringStartDate offering start date used as a reference
-   * to find all applications in the future.
+   * @param assessmentId assessment id.
    * @param entityManager used to execute the commands in the same transaction.
    */
   async rollbackOverawards(
-    studentId: number,
-    offeringStartDate: string,
+    assessmentId: number,
     entityManager: EntityManager,
   ): Promise<void> {
-    // Submitted applications are not allowed to have overlaps what makes possible
-    // to use the offering start date to order them.
-    // !Please note that, when doing local development and using
-    // !BYPASS_APPLICATION_SUBMIT_VALIDATIONS=true the application can be created with overlaps,
-    // !which is not a problem unless they have the same offering start date.
     const applicationsToRollback = await entityManager
       .getRepository(Application)
       .find({
-        select: { applicationNumber: true },
+        select: { applicationNumber: true, student: { id: true } },
+        relations: {
+          student: true,
+        },
         where: {
-          student: { id: studentId },
           currentAssessment: {
-            offering: { studyStartDate: MoreThanOrEqual(offeringStartDate) },
+            id: assessmentId,
```

Comment:
Can there be more than one application for a student-assessment?

Reply:
Nope, one is expected. This is a leftover, I will change it.

Reply:
Query adjusted.